lib/fizzbuzz_advanced.rb

```diff
@@ -0,0 +1,24 @@
+class FizzbuzzAdvanced
+  def initialize
+    @values = {
+      3 => "Fizz",
+      5 => "Buzz",
+      7 => "Sivv",
+      13 => "Grrr"
+    }
+  end
+
+  def calculate(num)
+    to_return = ""
+    output = ""
+
+    @values.each do |k, v|
+      output += v if num % k == 0
+    end
+
+    output.empty? ? to_return += num.to_s : to_return += output
```

Comment:
When you use `puts` to output, it will automatically append a new line. Is that the only reason `to_return` is in here? If so, I think you could just change line 19 to `output.empty? ? num.to_s : output`